shm/main.yml

```diff
@@ -0,0 +1,26 @@
+name: C++ with CMake with tests
+
+on:
+  push:
+    branches: [ homeworkOOP* ]
+  pull_request:
+    branches: [ homeworkOOP* ]
+
+jobs:
+  builder:
+
+    runs-on: ubuntu-18.04
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@master
+    - name: Make build directory
+      run: mkdir shm/build
+    - name: Compile
+      working-directory: shm/build
+      run: |
+        cmake ..
+        make
+    - name: Run tests
+      working-directory: shm/build
+      run: ctest -V   
```

Comment:
this file looks good, but it should be placed into .github/workflows directory. It doesn't have to be named main.yml. After that change, your project should be compiled and run by GitHub automatically :)